src/target/riscv/riscv.c

```diff
@@ -4010,74 +4010,85 @@ COMMAND_HANDLER(riscv_set_mem_access)
 	return ERROR_OK;
 }
 
-static int parse_ranges(struct list_head *ranges, const char *tcl_arg, const char *reg_type, unsigned int max_val)
-{
-	char *args = strdup(tcl_arg);
-	if (!args)
-		return ERROR_FAIL;
 
-	/* For backward compatibility, allow multiple parameters within one TCL argument, separated by ',' */
-	char *arg = strtok(args, ",");
-	while (arg) {
+static bool parse_csr_address(const char *reg_address_str, unsigned int *reg_addr)
```

Comment:
If i the description of those commands right - they still accept octal numbers. Given the nature of what we are dealing with here - I would forbid usage of octals to reduce risk of errors...

Though, now that I've said it - I think it would not hurt to add an actual check that the number does not start with zero.

Reply:
Good point about the octals. Better to leave these unsupported.

> I think it would not hurt to add an actual check that the number does not start with zero.

Sounds good.

Reply:
addressed